generated/niscope/niscope/session.py

```diff
@@ -1997,6 +1996,164 @@ def fetch(self, num_samples=None, relative_to=enums.FetchRelativeTo.PRETRIGGER,
+        digitizers.
+
+        Tip:
+        This method requires repeated capabilities. If called directly on the
+        niscope.Session object, then the method will use all repeated capabilities in the session.
+        You can specify a subset of repeated capabilities using the Python index notation on an
+        niscope.Session repeated capabilities container, and calling this method on the result.
+
+        Args:
+            array_meas_function (enums.ArrayMeasurement): The array measurement to perform.
+
+            timeout (hightime.timedelta, datetime.timedelta, or float in seconds): The time to wait in seconds for data to be acquired; using 0 for this
+                parameter tells NI-SCOPE to fetch whatever is currently available. Using
+                -1 for this parameter implies infinite timeout.
+
+            other_channel (str): The identifier for the "other channel" for multi-channel measurements such as Add Channels or Multiply Channels.
+
+
+        Returns:
+            wfm_info (list of WaveformInfo): Returns a list of class instances with the following timing and scaling
+                information about each waveform:
+
+                -  **relativeInitialX**—the time (in seconds) from the trigger to the
+                   first sample in the fetched waveform
+                -  **absoluteInitialX**—timestamp (in seconds) of the first fetched
+                   sample. This timestamp is comparable between records and
+                   acquisitions; devices that do not support this parameter use 0 for
+                   this output.
+                -  **xIncrement**—the time between points in the acquired waveform in
+                   seconds
+                -  **channel**-channel name this waveform was acquired from
+                -  **record**-record number of this waveform
+                -  **samples**—floating point array of samples. Length will be of actual samples acquired
+                -  **gain**—the gain factor of the given channel; useful for scaling
+                   binary data with the following formula:
+
+                voltage = binary data × gain factor + offset
+
+                -  **offset**—the offset factor of the given channel; useful for scaling
+                   binary data with the following formula:
+
+                voltage = binary data × gain factor + offset
+
+                Call _actual_num_wfms to determine the size of this array.
+
+        '''
+        # Set the fetch attributes
+        with _NoChannel(session=self):
+            if other_channel is not None:
+                self.meas_other_channel = other_channel
+
+        meas_wfm, wfm_info = self._fetch_array_measurement(array_meas_function, timeout)
+        record_length = int(len(meas_wfm) / len(wfm_info))
+
+        for i in range(len(wfm_info)):
+            start = i * record_length
+            end = start + record_length
```

Comment:
Shouldn't this be `wfm_info[i]._actual_samples`, like in `fetch()`?

Reply:
It's been long enough that I can't remember the details of why I made this change offhand. I'll check it out and see what's up with that.

Reply:
It looks like because originally the code was more different from fetch than it currently is, I had done stuff a little differently. There's no good reason for this change to be maintained, I'll go back to wfm_info[i]._actual_samples for consistency.